### PR TITLE
[WIP, not ready for review] Remove s3_direct_upload, add carrierwave_direct

### PR DESCRIPTION
**Pull request by citizen428**
## What type of PR is this? (check all applicable)

- [X] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, a note
on the devices and browsers this has been tested on, as well as any relevant
images for UI changes._

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Review by maestromac — COMMENTED** [review]
wow 8 dependencies removed

File: app/views/videos/new.html.erb [hunk 1]
```diff
@@ -1,49 +1,28 @@
...
-  form.upload-form {
-    background: #fdf9f3;
-    background: var(--theme-container-accent-background, #fdf9f3);
-    max-width: 590px;
-    margin: 20px auto;
-    padding: 40px 10px;
-    border: 2px solid #4e57ef;
-    box-shadow: 4px 5px 0px #4e57ef;
-  }
+form.new_video_uploader {
```
Comment (citizen428) [inline app/views/videos/new.html.erb:2]: Changed this to the default class name of the new helper.

File: app/views/videos/new.html.erb [hunk 2]
```diff
@@ -1,49 +1,28 @@
...
-      {%=o.name%}
-      <div class="progress"><div class="bar" style="width: 0%"></div></div>
-      <br/><br/><br/>
-    </div>
-    </script>
+  <%= direct_upload_form_for @uploader do |f| %>
+    <%= f.file_field :file,
+                     accept: VideoUploader.allowed_content_types.join(", "),
+                     onchange: "javascript:this.form.submit()",
+                     data: {
```
Comment (citizen428) [inline app/views/videos/new.html.erb:18]: Making use of the helpers in `app/javascript/packs/validateFileInputs.js`.

File: app/services/articles/create_with_video.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
+module Articles
+  class CreateWithVideo
```
Comment (citizen428) [inline app/services/articles/create_with_video.rb:2]: I decided to move this from `ArticleWithVideoCreationService` to something that is more in line with our current conventions. In the process I also added a `call` class method, so we went from 

```ruby
ArticleWithVideoCreationService.new(article_params, current_user).create!
```

to

```ruby
Articles::CreateWithVideo.call(params[:key], current_user)
````

**Pull request closed without merging**